spec/data-types-and-values.html

```diff
@@ -9,27 +9,109 @@ <h1>ECMAScript Language Types</h1>
       <h1>The Record Type</h1>
       <p>TBD</p>
 
-      <emu-clause id="sec-recordtostring" aoid="RecordToString">
-        <h1>RecordToString ( _argument_ )</h1>
-        <p>The abstract operation RecordToString takes argument _argument_ (a Record) and convertes _argument_ to a String. It performs the following steps when called:</p>
+      <emu-clause id="sec-immutable-types-record-tostring">
+        <h1>Record::toString ( _argument_ )</h1>
+        <p>The abstract operation Record::toString takes argument _argument_ (a Record) and convertes _argument_ to a String. It performs the following steps when called:</p>
         <emu-alg>
             1. Assert: Type(_argument_) is Record.
             1. Return *"[record]"*.
         </emu-alg>
       </emu-clause>
+
+      <emu-clause id="sec-immutable-types-record-equal">
+        <h1>Record::equal ( _x_, _y_, _elementEqual_ )</h1>
+        <p>The abstract operation Record::equal takes arguments _x_ (a Record), _y_ (a Record), and _elementEqual_ (a function object). _elementEqual_ will be invoked, with *undefined* as the receiver. It performs the following steps when called:</p>

```

Comment:
It seems like elementEqual should be an abstract closure instead of a JS function. Is there a reason it’d ever be user code?

Reply:
nope, no reason to expose this to user code. abstract closure works for me.